canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp

```diff
@@ -353,7 +368,23 @@ INT32 main(INT32 argc, CHAR* argv[])
                 }
                 lastKeyFrameTimestamp = frame.presentationTs;
             }
-            CHK_STATUS(putKinesisVideoFrame(streamHandle, &frame));
+            if (!intermittent) {
+                CHK_STATUS(putKinesisVideoFrame(streamHandle, &frame));
+                THREAD_SLEEP(HUNDREDS_OF_NANOS_IN_A_SECOND / DEFAULT_FPS_VALUE);
+            } else {
+                if (GETTIME() < runTill) {
+                    CHK_STATUS(putKinesisVideoFrame(streamHandle, &frame));
+                    THREAD_SLEEP(HUNDREDS_OF_NANOS_IN_A_SECOND / DEFAULT_FPS_VALUE);
+                } else {
```

Comment:
Trying to not dup this code. Set runTill to MAX_UINT64 or something for non intermittent.

Then just do checking 

if (GETTIME() < runTill) {
 putFrame
 Sleep();
} else {
// must be intermittent scenario with expired runTill
}

Reply:
Fixed.